> Ermm, the tests don't pass for me in py2 thinking
> 
> ```
> ======================================================================
> ERROR: testIndexedCollectionEqual (tests.mapped_struct.MappedListPackingTest)
> ----------------------------------------------------------------------
> Traceback (most recent call last):
>   File "/home/chris/workspace/sharedbuffers/tests/mapped_struct.py", line 1637, in testIndexedCollectionEqual
>     c = pack([1, 2.0])
>   File "/home/chris/workspace/sharedbuffers/tests/mapped_struct.py", line 1512, in pack
>     return self.PACKING_CLASS.unpack_from(a, 0)
>   File "sharedbuffers/mapped_struct.py", line 1054, in sharedbuffers.mapped_struct.mapped_list.unpack_from
>     index.frombytes(q)
> AttributeError: 'array.array' object has no attribute 'frombytes'
> 
> 
> ======================================================================
> ERROR: testIndexedCollectionError (tests.mapped_struct.MappedListPackingTest)
> ----------------------------------------------------------------------
> Traceback (most recent call last):
>   File "/home/chris/workspace/sharedbuffers/tests/mapped_struct.py", line 1606, in testIndexedCollectionError
>     c = self.pack([])
>   File "/home/chris/workspace/sharedbuffers/tests/mapped_struct.py", line 1512, in pack
>     return self.PACKING_CLASS.unpack_from(a, 0)
>   File "sharedbuffers/mapped_struct.py", line 1013, in sharedbuffers.mapped_struct.mapped_list.unpack_from
>     rv.frombytes(w)
> AttributeError: 'array.array' object has no attribute 'frombytes'
> 
> ...
> ```

Yes, sorry. It's called `fromstring` in py2, I need to do an if python3 there.

--- a/sharedbuffers/mapped_struct.py
+++ b/sharedbuffers/mapped_struct.py
@@ -1010,7 +1010,7 @@ class mapped_list(list):
                 rv = w.cast(dtype)
             else:
                 rv = array(dtype)
-                rv.frombytes(w)
+                rv.fromstring(w)
         elif dchar == b'q' or dchar == b'Q':
             if dchar == b'q':
                 dtype = 'l'
@@ -1026,7 +1026,7 @@ class mapped_list(list):
                 rv = q.cast(dtype)
             else:
                 rv = array(dtype)
-                rv.frombytes(q)
+                rv.fromstring(q)
         elif dchar == b'd':
             dtype = 'd'
             objlen, = _struct_l_Q.unpack(buf[offs:offs+8])
@@ -1037,7 +1037,7 @@ class mapped_list(list):
                 rv = q.cast(dtype)
             else:
                 rv = array(dtype)
-                rv.frombytes(q)
+                rv.fromstring(q)
         elif dchar == b't' or dchar == b'T':
             if dchar == b't':
                 dtype = 'l'
@@ -1051,7 +1051,10 @@ class mapped_list(list):
             offs += 8
             q = buf[offs:offs+itemsize*objlen]
             index = array(dtype)
-            index.frombytes(q)
+            if python3:
+                index.frombytes(q)
+            else:
+                index.fromstring(q)
 
             if idmap is None:
                 idmap = {}

